torch/__init__.py

```diff
@@ -917,6 +917,59 @@ def is_warn_always_enabled():
     """
     return _C._get_warnAlways()
 
+
+def _check_with(error_type, cond, message):
+    if not isinstance(cond, builtins.bool):
+        raise TypeError(f'cond must be a bool, but got {type(cond)}')
+
+    if cond:
+        return
+
+    # error_type must be a subclass of Exception and not subclass of Warning
+    assert issubclass(error_type, Exception) and not issubclass(error_type, Warning)
+
+    if message is None:
+        message_evaluated = (
+            'Expected cond to be True, but got False. (Could this error '
+            'message be improved? If so, please report an enhancement request '
+            'to PyTorch.)')
+
+    else:
+        if callable(message):
+            message_evaluated = str(message())
+        else:
+            message_evaluated = str(message)
```

Comment:
No, I don't want to accept the non-callable version. This was an intentional choice on the _prims version of check; it makes it to easy for people to start interpolating eagerly, which is a performance waste.